Sources/NIO/CircularBuffer.swift

```diff
@@ -589,7 +589,7 @@ extension CircularBuffer: RangeReplaceableCollection {
             self.removeSubrange(subrange)
         } else {
             var newBuffer: ContiguousArray<Element?> = []
-            let neededNewCapacity = self.count + newElements.count - subrangeCount + 1 /* always one spare */
+            let neededNewCapacity = self.count &+ newElements.count &- subrangeCount &+ 1 /* always one spare */
```

Comment:
I think in this case we probably want the checks as we're growing. I mean we will never be able to get to that many elements but I think it'd still be the right thing to do.

Reply:
fixed in a follow-up commit. Thanks